File: movies_repository/src/main/java/com/nikita/popularmovies/presentation/DefaultParams.java
package com.nikita.popularmovies.presentation;


public final class DefaultParams {
  public static void fun() {
    new Foo(1);
    new Foo(2, 3);
    new Foo(2, 8, 3);


    getNumber();
    getNumber(5);
    getNumber(5, 8);
  }

  public static final class Foo {
    public final int first;
    public final int second;

    public Foo(int second, int first) {
      this.first = first;
      this.second = second;
    }

    public Foo(int both) {
      this(both, both);
    }

    public Foo(int second, int multiplier, int first) {
      this(second * multiplier, first * multiplier);
    }
  }

  public static int getNumber() {
    return 4;
  }

  public static int getNumber(int number) {
    return getNumber() + number;
  }

  public static int getNumber(int number, int multiplier) {
    return getNumber(number * multiplier);
  }
}

File: movies_repository/src/main/kotlin/com/nikita/popularmovies/presentation/DefaultParams.kt
package com.nikita.popularmovies.presentation

fun foo() {
  Foo(first = 1)
  Foo(2, 3)
  Foo(2, 8, 3)

  FooSimple()
  FooSimple(first = 3)
  FooSimple(second = 3)
  FooSimple(second = 3, first = 2)

  getNumber()
  getNumber(5)
  getNumber(5, 8)
}

data class FooSimple(val first: Int = 4, val second: Int = 5)

class Foo(first: Int, multiplier: Int = 1, second: Int = 0) {
  val first: Int = first * multiplier
  val second: Int = if (second == 0) this.first else second * multiplier
}

@JvmOverloads
fun getNumber(number: Int = 0, multiplier: Int = 1) = (4 + number) * multiplier